-buildmode=pie is not supported for the mips arch

XiaodongLoong opened the pull request:
reference:
https://github.com/docker/cli/pull/2507
https://github.com/containerd/containerd/commit/4c99c81326f4026fb8c0b8c5e10542205d99c321

Signed-off-by: Xiaodong Liu <liuxiaodong@loongson.cn>

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

AkihiroSuda commented:
Can we split out the common part into a separate script?

XiaodongLoong commented:
> Can we split out the common part into a separate script?

Very Good advice!  I will try my best to make it!

XiaodongLoong commented:
@tianon I have modified the commit according to comment from AkihiroSuda , please review, Thanks!

AkihiroSuda commented on hack/dockerfile/install/rootlesskit.installer:
```diff
@@ -21,7 +21,7 @@ install_rootlesskit() {
 install_rootlesskit_dynamic() {
 	export ROOTLESSKIT_LDFLAGS="-linkmode=external" install_rootlesskit
-	export BUILD_MODE="-buildmode=pie"
+	export BUILD_MODE=${GO_BUILDMODE}
```
Can we use `BUILD_MODE` everywhere instead of introducing a new variable `GO_BUILDMODE`

XiaodongLoong replied:
I think `GO_BUILDMODE` is more specific and accurate. That change `GO_BUILDMODE` to `BUILD_MODE`  is also OK.

tianon replied:
Yeah, IMO this should go the other way and update everything to use the more specific `GO_BUILDMODE` name.

tianon reviewed (approved):
LGTM

AkihiroSuda commented:
needs rebase

XiaodongLoong commented:
@AkihiroSuda I use the conflict resolver provided by Github, is it OK？If not, I can close and reopen PR, rebase and subbmit a new commit againe.

AkihiroSuda commented:
Please squash

XiaodongLoong committed:
ldmode=pie is not supported for the mips arch
reference:
https://github.com/docker/cli/pull/2507
https://github.com/containerd/containerd/commit/4c99c81326f4026fb8c0b8c5e10542205d99c321

Signed-off-by: Xiaodong Liu <liuxiaodong@loongson.cn>

tianon commented:
LGTM :+1:

XiaodongLoong commented:
please merge, thanks!

XiaodongLoong commented:
@AkihiroSuda @tianon @thaJeztah

XiaodongLoong commented:
@AkihiroSuda I have squashed the commits, Thanks!

XiaodongLoong commented:
@tianon I rebased the commit according to AkihiroSuda, please review, I am so sorry for you have to review it again.

thaJeztah reviewed (approved):
LGTM
